R/ParamTree.R

```diff
@@ -0,0 +1,36 @@
+#' @title ParamSetTree
+#' @format \code{\link{R6Class}} object
+#'
+#' @description
+#' A \code{\link[R6]{R6Class}} to represent set of parameters in a tree form.
+#'
+#' @return [\code{\link{ParamSetTree}}].
+#' @family ParamHelpers
+#' @export
+ParamSetTree = R6Class("ParamSetTree",
+  inherit = ParamSet,
+  public = list(
+
+    # member variables
+    id = NULL,
+    handle = NULL,
+    flatval = NULL,
+    depend = NULL,
+
+    # constructor
+    initialize = function(id = NULL, handle = NULL, depend = NULL) {
+      self$id = id
```

Comment:
We need assertions for all arguments. Compare to [this](https://github.com/mlr-org/phng/blob/a7623fa7325d661a56c3e2e48d791bba4d4de217/R/ParamNode.R#L50)